examples/demo/gloo/galaxy/galaxy.py

```diff
@@ -0,0 +1,215 @@
+
+        data['a_color_index'] = (galaxy['temperature'] - t0) / (t1 - t0)
+        data['a_brightness'] = galaxy['brightness']
+        data['a_type'] = galaxy['type']
+
+        return data
+
+    def on_initialize(self, event):
+        # create a new shader program
+        self.program = gloo.Program(VERT_SHADER, FRAG_SHADER,
+                                    count=len(galaxy))
+
+        # load the star texture
+        self.texture = gloo.Texture2D(load_galaxy_star_image(),
+                                      interpolation='linear')
+        self.program['u_texture'] = self.texture
+
+        # construct the model, view and projection matrices
+        self.view = np.eye(4, dtype=np.float32)
+        transforms.translate(self.view, 0, 0, -5)
+        self.program['u_view'] = self.view
+
+        self.model = np.eye(4, dtype=np.float32)
+        self.program['u_model'] = self.model
+
+        self.program['u_colormap'] = colors
+
+        self.projection = perspective(45.0, self.width / float(self.height),
+                                      1.0, 1000.0)
+        self.program['u_projection'] = self.projection
+
+        # start the galaxy to some decent point in the future
+        galaxy.update(100000)
+        data = self.__create_galaxy_vertex_data()
+
+        # setup the VBO once the galaxy vertex data has been setup
+        # bind the VBO for the first time
+        self.data_vbo = gloo.VertexBuffer(data)
+        self.program.bind(self.data_vbo)
+
+        # setup blending
+        self.__setup_blending_mode()
+
+    def __setup_blending_mode(self):
+        # set the clear color to almost black
+        gloo.gl.glClearColor(0.0, 0.0, 0.03, 1.0)
+        gloo.gl.glDisable(gloo.gl.GL_DEPTH_TEST)
+        # use additive blend mode so that the
+        # particles overlap and intensify
+        gloo.gl.glEnable(gloo.gl.GL_BLEND)
+        gloo.gl.glBlendFunc(gloo.gl.GL_SRC_ALPHA, gloo.gl.GL_ONE)
+
+    def on_resize(self, event):
+        self.width, self.height = event.size
+        # setup the new viewport
+        gloo.set_viewport(0, 0, self.width, self.height)
+        # recompute the projection matrix
+        self.projection = perspective(45.0, self.width / float(self.height),
+                                      1.0, 1000.0)
+        self.program['u_projection'] = self.projection
+
+    def on_draw(self, event):
+        # update the galaxy
+        galaxy.update(50000)  # in years !
+
+        # recreate the numpy array that will be sent as the VBO data
+        data = self.__create_galaxy_vertex_data()
+        # update the VBO
+        self.data_vbo.set_data(data)
+        # bind the VBO to the GL context
+        self.program.bind(self.data_vbo)
+
+        # clear the screen and render
+        gloo.clear(color=True, depth=True)
+        self.program.draw('points')
+
+if __name__ == '__main__' and sys.flags.interactive == 0:
+    c = Canvas()
+    c.show()
+    app.run()
```

Comment:
I think Eric is saying the code should be like this:

```python
canvas = Canvas()
canvas.show()

if __name__ == '__main__' and sys.flags.interactive == 0:
    app.run()
```

Note how just `app.run()` is inside the conditional. This way, you can run this interactively in IPython and access `canvas`.